composer/utils/checkpoint.py

```diff
@@ -275,12 +290,24 @@ def read_data(self, plan: LoadPlan, planner: LoadPlanner):
                         download_object_or_file(object_name, file_destination, self.object_store)
                         log.debug(f'Finished downloading {relative_file_path} to {file_destination}.')
             except Exception as e:
-                # PyTorch will capture any exception of this function,
-                # and dist.all_gather_objects(exception) before raising it.
-                # If that all_gather_objects fails, the exception is never visible to user.
-                # We immediately print the exception to avoid that situation.
                 log.error(f'Exception {type(e)} raised during downloading: {str(e)}')
-                raise e
+                download_error = True
+
+        # PyTorch will capture any exception of this function,
+        # and dist.all_gather_objects(exception) before raising it.
+        # If that all_gather_objects fails, the exception is never visible to user.
+        # We immediately kill the process and print the exception
+        download_error_tensor = torch.tensor(1 if download_error else 0)
+        error_by_rank = dist.all_gather(download_error_tensor)
+        failed_ranks = []
+        for rank, error in enumerate(error_by_rank):
+            if error.item() > 0:
+                failed_ranks.append(rank)
+                download_error = True
+
+        if download_error:
+            self.terminate_all_processes()
+            log.error(f'Ranks {failed_ranks} failed to download. Terminating all processes to end the run.',)
```

Comment:
i think you need to `log.error` first,  then `self.terminate_all_processes()` ?

Reply:
oops 😆